File: 백준/Bronze/32297. 문자열을 만들어요/문자열을 만들어요.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N = int(input())
    g = 'gori'
    S = input()
    for n in range(N):
        if S[n:n+4] == g:
            print('YES')
            return
    print('NO')


if __name__ == '__main__':
    main()

File: 백준/Bronze/32297. 문자열을 만들어요/문자열을 만들어요.c
#include <stdio.h>
#include <string.h>

int main() {
    int N;
    char S[11]; 
    char g[] = "gori";
    
    scanf("%d", &N);
    scanf("%s", S);
    
    for (int n = 0; n <= N - 4; n++) {
        if (strncmp(&S[n], g, 4) == 0) {
            printf("YES\n");
            return 0;
        }
    }
    
    printf("NO\n");
    return 0;
}